module/dkg/broker.go

```diff
@@ -18,37 +18,43 @@ import (
 	"github.com/onflow/flow-go/model/flow"
 	"github.com/onflow/flow-go/model/messages"
 	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/retrymiddleware"
 )
 
-// retryMaxPublish is the maximum number of times the broker will attempt to broadcast
-// a message or publish a result
-const retryMaxPublish = 8
+const (
+	// retryMaxPublish is the maximum number of times the broker will attempt to broadcast
+	// a message or publish a result
+	retryMaxPublish = 8
 
-// retryMaxRead is the max number of times the broker will attempt to read messages
-const retryMaxRead = 3
+	// retryMaxRead is the max number of times the broker will attempt to read messages
+	retryMaxRead = 3
 
-// retryMilliseconds is the number of milliseconds to wait between retries
-const retryMilliseconds = 1000 * time.Millisecond
+	// retryMilliseconds is the number of milliseconds to wait between retries
+	retryMilliseconds = 1000 * time.Millisecond
+
+	// retryMaxConsecutiveFailures is the number of consecutive failures allowed before we attempt with fallback client
+	retryMaxConsecutiveFailures = 2
+)
 
 // Broker is an implementation of the DKGBroker interface which is intended to
 // be used in conjuction with the DKG MessagingEngine for private messages, and
 // with the DKG smart-contract for broadcast messages.
 type Broker struct {
 	sync.Mutex
```

Comment:
It's a bit confusing why it has an embedded lock and then uses `unit` for locking as well.

Reply:
https://github.com/onflow/flow-go/pull/1556/commits/57c05d77a5b4485866c7036db4ad16e4ec2a9588